app/receiver/index.ts

```diff
@@ -0,0 +1,177 @@
+/*!
+ * Copyright © 2022 United States Government as represented by the Administrator
+ * of the National Aeronautics and Space Administration. No copyright is claimed
+ * in the United States under Title 17, U.S. Code. All Other Rights Reserved.
+ *
+ * SPDX-License-Identifier: NASA-1.3
+ */
+
+import {
+  ListUsersInGroupCommand,
+  CognitoIdentityProviderClient,
+} from '@aws-sdk/client-cognito-identity-provider'
+import { S3Client, GetObjectCommand } from '@aws-sdk/client-s3'
+import { DynamoDBAutoIncrement } from '@nasa-gcn/dynamodb-autoincrement'
+import { DynamoDBClient } from '@aws-sdk/client-dynamodb'
+import { DynamoDBDocument } from '@aws-sdk/lib-dynamodb'
+import { SESClient, SendEmailCommand } from '@aws-sdk/client-ses'
+import type { S3Event } from 'aws-lambda'
+
+import type { Source } from 'mailparser'
+import { simpleParser } from 'mailparser'
+
+import {
+  emailIsAutoReply,
+  subjectIsValid,
+  formatAuthor,
+} from '../routes/circulars/circulars.lib'
+
+import { tables } from '@architect/functions'
+import { extractAttributeRequired, extractAttribute } from '~/lib/cognito'
+
+const s3 = new S3Client({})
+const client = new CognitoIdentityProviderClient({})
+const sesClient = new SESClient({})
+const dynamodbClient = new DynamoDBClient({})
+
+const USER_POOL_ID = process.env.USER_POOL_ID
+
+export async function handler(event: S3Event) {
+  const bucket = event.Records[0].s3.bucket.name
+  const key = decodeURIComponent(
+    event.Records[0].s3.object.key.replace(/\+/g, ' ')
+  )
+  const getObjectParams = {
+    Bucket: bucket,
+    Key: key,
+  }
+  // Get data from S3
+  const object = await s3.send(new GetObjectCommand(getObjectParams))
+  if (!object.Body) throw new Error('S3 object has no body')
+
+  const parsed = await simpleParser(object.Body as unknown as Source)
```

Comment:
Why do you need to override the type inference?

Reply:
I was getting errors at the time that the Body couldn't be parsed, but it looks like it is working with just the cast to Source. I am removing the `as unknown` part